examples/sample_dir/assignment_inside_condition.py

```diff
@@ -0,0 +1,2 @@
+if spam = 42:  # Error on this line: assignment (`=`) instead of equality (`==`)
```

Comment:
It's great that you created an example directory. Two comments:

1. This `sample_dir` shouldn't be under `examples` since it's not an example. You can move this under the `tests/fixtures` folder.
2. You don't need to include that many files; maybe just one or two files per "level" and one or two levels of nested directories.